demo/cpp/model_peeker.cc

```diff
@@ -0,0 +1,99 @@
+#include <iostream>
+#include <cstdio>
+#include <fstream>
+#include <numeric>
+#include <algorithm>
+#include <functional>
+#include <vector>
+#include <limits>
+#include <stdexcept>
+#include <dlr.h>
+#include "dmlc/logging.h"
+
+/*! \brief Prints model metadata using DLR C-API.
```

Comment:
I don't have strong preference. Do you want to add it to the test pipeline (using gtest or could a script to build and run it somehow). Otherwise, if ppl change DLR but forgot to update this demo, it may just fail later when other ppl use it.

Reply:
The demo is using only core APIs that are intended to be alway backward-compatible. Will add C-API gtests in following PRs.